feat: gating test

just-mitch opened the pull request:
Main feature of this branch (unlike the branch name indicates) is the `gating_passive.test.ts`.

This test:
- updates the aztec network deployment, allowing validators to use each other as boot nodes
- applies the "network-requirements" network shaping
- permanently disables the boot node
- runs 3 epochs during which it:
  - kills 25% of the validators
  - asserts that we miss less than 50% of slots

Other work in this branch includes:
- allows sequencers to start building as soon as they see they can propose in the next L1 slot
- add `ignoreDroppedReceiptsFor` TX wait options
  - this allows sending a TX to one node, and awaiting it on another since we need time for p2p propagation
  - we need this since we have shifted the PXE to point at the top-level validator service, which load balances across individuals
  - this may help with #9613
- scalable loki deployment for prod
- more visible logging for core sequencer operations
- better error handling during the setup of l2 contracts
- better error handling in the pxe
- rename the network shaping charts to "aztec-chaos-scenarios"

Fix #9713 
Fix #9883

just-mitch commented on spartan/aztec-chaos-scenarios/values/prover-failure.yaml:
```diff
@@ -0,0 +1,8 @@
+# Simulates congested network conditions
+# High latency, limited bandwidth, packet loss
```
Will remove

just-mitch commented on spartan/aztec-chaos-scenarios/values/validator-kill.yaml:
```diff
@@ -0,0 +1,8 @@
+# Simulates congested network conditions
+# High latency, limited bandwidth, packet loss
```
will remove

just-mitch commented on spartan/aztec-network/templates/_helpers.tpl:
```diff
@@ -78,7 +78,7 @@ http://{{ include "aztec-network.fullname" . }}-boot-node-0.{{ include "aztec-ne
 {{- if .Values.validator.externalTcpHost -}}
 http://{{ .Values.validator.externalTcpHost }}:{{ .Values.validator.service.nodePort }}
 {{- else -}}
-http://{{ include "aztec-network.fullname" . }}-validator-0.{{ include "aztec-network.fullname" . }}-validator.{{ .Release.Namespace }}.svc.cluster.local:{{ .Values.validator.service.nodePort }}
```
Adjusts `validatorUrl` to load balance over all validators.

just-mitch commented on spartan/aztec-network/templates/validator.yaml:
```diff
@@ -19,39 +19,73 @@ spec:
+                      # the PXE has its node set to the the validator service.
+                      # and that's all we need to know to bootstrap,
+                      # since it will get a good node ENR from whatever node the PXE connects to.
+                      echo "{{ include "aztec-network.pxeUrl" . }}" > /shared/pxe_url
```
As noted in the comment above, this looping is really just a safety/sanity check, since as long as one validator is up, the PXE *should* be routed to it.

just-mitch commented on spartan/aztec-network/values/4-validators-with-metrics.yaml:
smallest set where we can kill 25%.

just-mitch commented on spartan/metrics/install-prod.sh:
```diff
@@ -3,4 +3,4 @@ set -eu
 
 cd "$(dirname "${BASH_SOURCE[0]}")"
 
-helm upgrade metrics . -n metrics --values "./values/prod.yaml" --install --create-namespace --atomic $@
```
Happy to revert this change, but removing atomic was necessary to try to debug.

just-mitch commented on yarn-project/protocol-contracts/src/protocol_contract_data.ts:
will delete.

just-mitch commented on yarn-project/sequencer-client/src/sequencer/sequencer.ts:
```diff
@@ -167,11 +167,11 @@ export class Sequencer {
-      [SequencerState.PUBLISHING_BLOCK_TO_PEERS]: 5 + this.maxTxsPerBlock * 2, // if we take 5 seconds to create block, then 4 transactions at 2 seconds each
-      [SequencerState.WAITING_FOR_ATTESTATIONS]: 5 + this.maxTxsPerBlock * 2 + 3, // it shouldn't take 3 seconds to publish to peers
-      [SequencerState.PUBLISHING_BLOCK]: 5 + this.maxTxsPerBlock * 2 + 3 + 5, // wait 5 seconds for attestations
+      [SequencerState.WAITING_FOR_TXS]: 5,
```
Had to bump these times, as we were often trying to transition to "WAITING_FOR_TXS" at ~3.5 seconds, which was resulting in no blocks getting produced, which resulted in reorgs due at least in part to lack of support for https://github.com/AztecProtocol/aztec-packages/issues/9404

just-mitch commented on yarn-project/sequencer-client/src/sequencer/sequencer.ts:
```diff
@@ -573,19 +573,19 @@ export class Sequencer {
 
       this.isFlushing = false;
-      this.log.verbose('Collecting attestations');
+      this.log.info('Collecting attestations');
```
@ludamad I'm actually not seeing these logs under in loki even with debug on. Thoughts? I'm not sure if it is all `verbose` logs, or what. That was part of the impetus for upping the level.

just-mitch replied:
I filed https://github.com/AztecProtocol/aztec-packages/issues/9958

ludamad replied:
Unfortunately I am not sure. We should write good tests along with moving to a non-node debugger (winston? pino?)

just-mitch committed:
feat: gating_passive.test.ts

This test:
- updates the aztec network deployment, allowing validators to use each other as boot nodes
- applies the "network-requirements" network shaping
- permanently disables the boot node
- runs 3 epochs during which it:
  - kills 25% of the validators
  - asserts that we miss less than 50% of slots

Other work in this branch includes:
- add `ignoreDroppedReceiptsFor` TX wait options
  - this allows sending a TX to one node, and awaiting it on another since we need time for p2p propagation
  - we need this since we have shifted the PXE to point at the top-level validator service, which load balances across individuals
  - this may help with #9613
- scalable loki deployment for prod
- more visible logging for core sequencer operations
- better error handling during the setup of l2 contracts
- better error handling in the pxe
- rename the network shaping charts to "aztec-chaos-scenarios"

just-mitch committed:
fix conflict

spalladino commented on yarn-project/sequencer-client/src/sequencer/sequencer.ts:
I'm not a fan of bumping so many `verbose` to `info`, it'll make all tests so much more verbose. But if we need this inorder to see those logs in k8s tests, I'm fine with it until we move to a better logger.

just-mitch replied:
Yeah agreed that the more sophisticated logger is the way.

I bumped it because I wasn't seeing them, and because I figured that in production, you'd want to see when you're producing a block.

spalladino commented on yarn-project/sequencer-client/src/sequencer/sequencer.ts:
```diff
@@ -392,14 +393,13 @@ export class Sequencer {
       );
       return;
     }
-    const secondsIntoSlot = getSecondsIntoSlot(this.l1GenesisTime, this.aztecSlotDuration);
```
Why not pass in the current slot number, and keep the `getSecondsIntoSlot` call here? Seems like it would reduce some verbosity by not having to manually call `getSecondsIntoSlot` before each `setState`.

spalladino reviewed (commented):
Looked into all non-k8s stuff

Maddiaa0 commented on yarn-project/end-to-end/src/spartan/utils.ts:
```diff
@@ -1,15 +1,19 @@
 import { promisify } from 'util';
 import { z } from 'zod';
 
+import type { RollupCheatCodes } from '../../../aztec.js/src/utils/cheat_codes.js';
```
dodgy update import here

stevenplatt commented on spartan/aztec-chaos-scenarios/values.yaml:
```diff
@@ -1,3 +1,6 @@
+
 global:
   # When deploying, override the namespace to where spartan will deploy to, this will apply all chaos experiments to all pods within that namespace
   # run deployment with --values global.namespace=your-namespace
```
Nice, thanks for including this note. It wasn't clear to me at first why the values files used different namespace refs.

stevenplatt commented on spartan/aztec-network/values/48-validators.yaml:
```diff
@@ -9,7 +9,6 @@ telemetry:
   otelCollectorEndpoint: http://metrics-opentelemetry-collector.metrics:4318
 
 validator:
-  debug: "aztec:*,-aztec:avm_simulator:*,-aztec:libp2p_service"
```
Does this level of logging causing stability issues? (for my own understanding)

just-mitch replied:
We've seen issues in e2e tests where excessive logging messes up tests. I haven't seen it in network tests as far as I am aware. 

However, this was actually reducing the level of logs, since the default level in values.yaml has more negative filters.

Maddiaa0 reviewed (approved):
turbo nits

just-mitch commented on yarn-project/end-to-end/src/spartan/utils.ts:
```diff
@@ -181,42 +221,164 @@ export async function installChaosMeshChart({
+  });
+}
+
+export async function awaitL2BlockNumber(
+  rollupCheatCodes: RollupCheatCodes,
+  blockNumber: bigint,
+  timeoutSeconds: number,
+  logger: Logger,
+) {
+  logger.info(`Waiting for L2 Block ${blockNumber}`);
+  let tips = await rollupCheatCodes.getTips();
+  const endTime = Date.now() + timeoutSeconds * 1000;
+  while (tips.pending < blockNumber && Date.now() < endTime) {
+    logger.info(`At L2 Block ${tips.pending}`);
+    await sleep(1000);
+    tips = await rollupCheatCodes.getTips();
+  }
+  logger.info(`Reached L2 Block ${tips.pending}`);
+}
```
filed https://github.com/AztecProtocol/aztec-packages/issues/10074

just-mitch commented on yarn-project/pxe/src/pxe_service/pxe_service.ts:
```diff
@@ -496,10 +496,19 @@ export class PXEService implements PXE {
+        );
+        return new TxProvingResult(privateExecutionResult, publicInputs, clientIvcProof!);
+      })
+      .catch(err => {
+        this.log.error(err);
+        throw err;
+      });
```
Yes, it does. But agreed on context. Filed https://github.com/AztecProtocol/aztec-packages/issues/10075

stevenplatt commented on spartan/aztec-chaos-scenarios/templates/network-shaping.yaml:
```diff
@@ -57,7 +57,7 @@ metadata:
-    {{- include "network-shaping.labels" . | nindent 4 }}
+    {{- include "aztec-chaos-scenarios.labels" . | nindent 4 }}
   annotations:
     "helm.sh/resource-policy": keep
```
I understand this annotation to mean that this resource is left behind when he helm chart is uninstalled. Is this intentional? 

Would applying `"helm.sh/hook": post-install,post-upgrade` and `"helm.sh/hook-weight": "<some-weight>"` be sufficient for the scenarios?

just-mitch commented on spartan/aztec-network/templates/setup-l2-contracts.yaml:
```diff
@@ -32,7 +32,7 @@ spec:
           env:
             - name: PXE_URL
-              value: {{ include "aztec-network.pxeUrl" . | quote }}
+              value: {{ include "aztec-network.bootNodeUrl" . | quote }}
```
Just to simplify, since the boot node is a PXE.
